contracts/amm.cairo

```diff
@@ -119,13 +141,21 @@ func do_trade{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
 
         // 3) Calculate new volatility, calculate trade volatility
         
-        let (current_pool_balance) = get_pool_available_balance(lptoken_address);
-        assert_nn_le(1, current_pool_balance);
+        let (current_pool_balance) = get_unlocked_capital(
+            lptoken_address=lptoken_address
+        );
         with_attr error_message("not enough assets in pool to fulfill requested trade") {
-            assert_nn_le(option_size_in_pool_currency, current_pool_balance);
+            assert_uint256_le(option_size_in_pool_currency, current_pool_balance);
+        }
+
+        if (option_type == OPTION_CALL) {
```

Comment:
> Why not 

Because it saves steps+memory holes+syscalls, no point in fetching from storage_var when we have it already. Yes only slightly (steps=53563 -> steps=53255).

Added a new function to helpers that does this and switched to using it in a few places.

Reply:
:+1: